src/components/Home.tsx

```diff
@@ -1,21 +1,57 @@
-import { useEffect } from 'react';
+import { useEffect, useState } from 'react';
 import { useLocation, useHistory } from 'react-router-dom';
+
 import { UserInfo } from '../type';
-import { auth } from '../firebase';
-import '../App.css'
+import { auth, db } from '../firebase';
+import '../App.css';
+import { Modal } from './Modal';
 
 export const Home = () => {
   const location = useLocation<UserInfo>();
   const state = location?.state;
   const history = useHistory();
 
+  const [otherUsers, setOherUsers] = useState<UserInfo[]>([]);
+  const [isShowModal, setIsShowModal] = useState(false);
+  const [modalUser, setModalUser] = useState<UserInfo>();
+
   useEffect(() => {
     auth.onAuthStateChanged((user) => {
-      if (!user) return history.push('/Login');
+      if (!user) history.push('/Login');
+      else getOtherUserInfo();
     });
-  });
+  }, []);
+
+  const openModal = (userInfo: UserInfo): void => {
+    setModalUser(userInfo);
+    setIsShowModal(true);
+  };
+
+  const getOtherUserInfo = async (): Promise<void> => {
+    try {
+      const myUid = await auth.currentUser?.uid;
+      if (myUid)
```

Comment:
if文に関して
すごく細かい点なのですが、異常系の処理を先に持ってきた方が可読性が高くなります。
今回の場合、mUidがfalseの時の処理を先に記述し、returnやエラー通知などで処理を止める実装の方がより可読性が高いです。
```suggestion
      if (!myUid) {
        // エラー処理など
      }
      // 正常系の処理
      await db,,,,,
```